File: Minimum Number.cpp
class Solution{
public:
    int minimumNumber(int n,vector<int> &arr){
        int left =0, right=n-1;
        while(left<right){
            if(arr[left]>=arr[right]) arr[left]=arr[left]%arr[right];
            else if(arr[right]>=arr[left]) arr[right]=arr[right]%arr[left];
            
            if(arr[left]==0) left++;
            else if(arr[right]==0) right--;
        }
        return arr[left];
    }
};

File: Minimum Number.java
class Solution {
    public static int minimumNumber(int n, int[] arr) {
        // code here
        int i=0,res=0;
        for(i=0;i<n;i++)
        {
            res=gcd(res,arr[i]);
        }
        return res;
    }
    public static int gcd(int a,int b)
    {
        if(b==0)
        return a;
        else
        return gcd(b,a%b);
    }
}
